### PR TITLE
HPCC-26306 add remaining logical file meta information methods

### PR DESCRIPTION
**Pull request by Michael-Gardner**
## Type of change:
- [ ] This change is a bug fix (non-breaking change which fixes an issue).
- [x] This change is a new feature (non-breaking change which adds functionality).
- [x] This change improves the code (refactor or other change that does not change the functionality)
- [ ] This change fixes warnings (the fix does not alter the functionality or the generated code)
- [ ] This change is a breaking change (fix or feature that will cause existing behavior to change).
- [ ] This change alters the query API (existing queries will have to be recompiled)

## Checklist:
- [x] My code follows the code style of this project.
  - [ ] My code does not create any new warnings from compiler, build system, or lint.
- [x] The commit message is properly formatted and free of typos.
  - [ ] The commit message title makes sense in a changelog, by itself.
  - [ ] The commit is signed.
- [ ] My change requires a change to the documentation.
  - [ ] I have updated the documentation accordingly, or...
  - [ ] I have created a JIRA ticket to update the documentation.
  - [ ] Any new interfaces or exported functions are appropriately commented.
- [x] I have read the CONTRIBUTORS document.
- [x] The change has been fully tested:
  - [ ] I have added tests to cover my changes.
  - [ ] All new and existing tests passed.
  - [ ] I have checked that this change does not introduce memory leaks.
  - [ ] I have used Valgrind or similar tools to check for potential issues.
- [ ] I have given due consideration to all of the following potential concerns:
  - [ ] Scalability
  - [ ] Performance
  - [ ] Security
  - [ ] Thread-safety
  - [ ] Cloud-compatibility
  - [ ] Premature optimization
  - [ ] Existing deployed queries will not be broken
  - [ ] This change fixes the problem, not just the symptom
  - [ ] The target branch of this pull request is appropriate for such a change.
- [ ] There are no similar instances of the same problem that should be addressed
  - [ ] I have addressed them here
  - [ ] I have raised JIRA issues to address them separately
- [ ] This is a user interface / front-end modification
  - [ ] I have tested my changes in multiple modern browsers
  - [ ] The component(s) render as expected

## Smoketest:
- [ ] Send notifications about my Pull Request position in Smoketest queue.
- [ ] Test my draft Pull Request.

### CODE REVIEW

**Comment by wangkx** [comment]
@Michael-Gardner I got 404 errors when I click the links of your commits.

**Review by wangkx — COMMENTED** [review]
please see the comments

File: esp/services/ws_dali/ws_daliservice.cpp [hunk 1]
```diff
@@ -539,3 +539,370 @@ bool CWSDaliEx::onGetProtectedList(IEspContext& context, IEspGetProtectedListReq
...
+
+bool CWSDaliEx::onDFSReplication(IEspContext& context, IEspDFSReplicationRequest& req, IEspResultResponse& resp)
+{
+    try
+    {
+        checkAccess(context);
+
+        const char* clusterMask = req.getClusterMask();
+        if (isEmptyString(clusterMask))
+            throw makeStringException(ECLWATCH_INVALID_INPUT, "Cluster Mask not specified.");
```
[the remark this answers is not in the document]
Comment (Michael-Gardner) [reply esp/services/ws_dali/ws_daliservice.cpp:551]: Not required. If it's a nullptr it is treated just like a wildcard. Fixing

File: esp/services/ws_dali/ws_daliservice.cpp [hunk 2]
```diff
@@ -539,3 +539,370 @@ bool CWSDaliEx::onGetProtectedList(IEspContext& context, IEspGetProtectedListReq
...
+
+        const char* lfnMask = req.getLogicalNameMask();
+        if (isEmptyString(lfnMask))
+            throw makeStringException(ECLWATCH_INVALID_INPUT, "Logical File Mask not specified.");
```
[the remark this answers is not in the document]
Comment (Michael-Gardner) [reply esp/services/ws_dali/ws_daliservice.cpp:555]: Same as above. if nullptr it is treated like no lfnmask exists and it's not appended to the findXPath stringbuffer.

File: esp/services/ws_dali/ws_daliservice.cpp [hunk 3]
```diff
@@ -539,3 +539,370 @@ bool CWSDaliEx::onGetProtectedList(IEspContext& context, IEspGetProtectedListReq
...
+    {
+        checkAccess(context);
+
+        const char* name = req.getName();
+        if (isEmptyString(name))
+            throw makeStringException(ECLWATCH_INVALID_INPUT, "Name not specified.");
+
+        bool fix = req.getFix();
+        if (req.getFix_isNull())
+            throw makeStringException(ECLWATCH_INVALID_INPUT, "Fix not specified.");
```
[the remark this answers is not in the document]
Comment (Michael-Gardner) [reply esp/services/ws_dali/ws_daliservice.cpp:774]: No it's not, I've gone ahead and modified it to act accordingly if the user doesn't set a value for fix.

File: esp/services/ws_dali/ws_daliservice.cpp [hunk 4]
```diff
@@ -539,3 +539,370 @@ bool CWSDaliEx::onGetProtectedList(IEspContext& context, IEspGetProtectedListReq
...
+{
+    try
+    {
+        checkAccess(context);
+
+        Owned<IUserDescriptor> userDesc = createUserDesc(context);
+
+        const char* name = req.getName();
+        if (isEmptyString(name))
+            throw makeStringException(ECLWATCH_INVALID_INPUT, "Name not specified.");
```
[the remark this answers is not in the document]
Comment (Michael-Gardner) [reply esp/services/ws_dali/ws_daliservice.cpp:581]: No name is required, will default to a wildcard in the underlying 'normalizeFileNames function'.  Fixing

**Comment by stuartort** [comment]
@wangkx , please re-review

**Review by wangkx — COMMENTED** [review]
@Michael-Gardner the changes look fine. But, the following review comments are not resolved:
https://github.com/hpcc-systems/HPCC-Platform/pull/15365/commits/fd49713d58c660bf1d958cf350c7049f5da1215c#r712486348
https://github.com/hpcc-systems/HPCC-Platform/pull/15365/commits/fd49713d58c660bf1d958cf350c7049f5da1215c#r712491506

**Comment by stuartort** [comment]
@Michael-Gardner , please see Kevin's comment about non-resolved items

**Comment by stuartort** [comment]
@Michael-Gardner , @wangkx  this looks stalled.  Once complete, do we want it still targetting 8.4 or should we target 8.6

**Comment by Michael-Gardner** [comment]
I think back into 8.4 might be appropriate. It's just fleshing out the existing feature that already exists

**Pull request closed without merging**